drivers/tapdisk-image.c

```diff
@@ -287,6 +289,53 @@ tapdisk_image_close_chain(struct list_head *list)
 		tapdisk_image_close(image);
 }
 
+static int
+find_parent_nbd_path(int prt_devnum, char **prt_nbd_path)
+{
+	int err;
+	char *pattern;
+	glob_t glbuf = { 0 };
+
+	/* This is glob pattern not a regex */
+	err = asprintf(&pattern, "%s/nbd[0-9]*.%d", BLKTAP2_CONTROL_DIR, prt_devnum);
+	if (err == -1) {
+		errno =-errno;
+		return err;
```

Comment:
I think you meant 'return -errno' here (or 'err = -errno' followe by 'return err' if you really want to), otherwise this is equivalent to 'return -1', but everywhere else in this function 'err' is set to '-errno' and then returned.

Reply:
Yes, I think you're correct, or at least intended `err = - errno;` anyway.